packages/client/katello-pull-provider-migrate/katello-pull-provider-migrate.sh

```diff
@@ -0,0 +1,82 @@
+#!/bin/sh
+
+# set SYSCONFDIR to /etc if it is not set
+if [ -z "$SYSCONFDIR" ]; then
+    SYSCONFDIR=/etc
+fi
+
+# see if /etc/rhsm/rhsm.conf exists
+RHSMCONF_PATH=$SYSCONFDIR/rhsm/rhsm.conf
+if [ -f $RHSMCONF_PATH ]; then
+    echo "Getting configuration from $RHSMCONF_PATH..."
+    # get cert-file and key-file from rhsm.conf
+    CONSUMER_CERT_DIR=$(cat ${RHSMCONF_PATH} | grep consumerCertDir | cut -f3 -d ' ')
+    CERT_FILE=$CONSUMER_CERT_DIR/cert.pem
+    KEY_FILE=$CONSUMER_CERT_DIR/key.pem
+    # get server name from rhsm.conf
+    SERVER_NAME=$(cat $RHSMCONF_PATH | grep -m1 'hostname = ' | cut -f3 -d ' ')
+else
+    echo "$RHSMCONF_PATH not found; exiting"
+    exit 1
+fi
+
+# fail if no server name, cert file or key file
+if [ -z "$SERVER_NAME" ] || [ -z "$CONSUMER_CERT_DIR" ]; then
+    echo "Unable to determine hostname or consumerCertDir from $SYSCONFDIR/rhsm/rhsm.conf; exiting"
+    exit 1
+fi
+
+# fail if client is not registered to a Satellite
+if [ $(grep -c "prefix = /rhsm" $RHSMCONF_PATH) -eq 0 ]; then
+    echo "Client is not registered to Katello; exiting"
+    exit 2
+fi
+
+# see if /etc/yggdrasil/config.toml exists
+CONFIGTOML=$SYSCONFDIR/yggdrasil/config.toml
+if [ -f $CONFIGTOML ]; then
+    # make a backup of CONFIGTOML
+    cp $CONFIGTOML $CONFIGTOML.bak
+    echo "Writing configuration to $CONFIGTOML..."
+    # find the line starting with 'broker = ' and overwrite it with the broker url
+    sed -i "s/broker = .*/broker = [\"tcp:\/\/$SERVER_NAME:1883\"]/" $CONFIGTOML
```

Comment:
Instead of modifying an existing line or appending it to the file if there is none, wouldn't it be easier to just write the whole file from scratch? Currently the only untouched line from the upstream config is log level